### PR TITLE
Allow docstrings on defspec

### PR DESCRIPTION
**Pull request by dball**
Assuming this is desirable and a good approach, I'll submit something
similar for defunion and defenum.

Note this incidentally cleans up trailing whitespace, which I'm happy to
submit separately or not at all, if the diff foo is annoying.

### CODE REVIEW

File: src/spark/spec_tacular.clj [hunk 1]
```diff
@@ -623,7 +623,7 @@
 
 (defmacro defspec
```
Comment (calvis) [inline src/spark/spec_tacular.clj]: can you update the documentation for `defspec` to include the optional docstring after `Name`

**Comment by dball** [comment]
Absolutely. Approach is reasonable then?

Should the spec record get a doc field, or let it just be one of them loosey-goosey free fields?

**Comment by dball** [comment]
I reckon I'm satisfied with this if you are, @calvis

File: src/spark/spec_tacular.clj [hunk 2]
```diff
@@ -623,16 +623,17 @@
...
   "Defines a spec-tacular spec type.
-  
+
   ```
   (defspec Name
-    [field-name arity type option ...]
+    [field-name docstring? arity type option ...]
```
Comment (calvis) [inline src/spark/spec_tacular.clj]: I think this is not where it goes?  should be next to `Name`

**Pull request merged**